standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/client/builder/GetPartitionsRequestBuilder.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore.client.builder;
+
+import org.apache.hadoop.hive.metastore.api.GetPartitionsFilterSpec;
+import org.apache.hadoop.hive.metastore.api.GetPartitionsRequest;
+import org.apache.hadoop.hive.metastore.api.GetProjectionsSpec;
+
+import java.util.List;
+
+public class GetPartitionsRequestBuilder {
+    private String catName = null;
+    private String dbName = null;
+    private String tblName = null;
+    private boolean withAuth = true;
+    private String user = null;
+    private List<String> groupNames = null;
+    private GetProjectionsSpec projectionSpec = null;
+    private GetPartitionsFilterSpec filterSpec = null;
+    private List<java.lang.String> processorCapabilities = null;
+    private String processorIdentifier = null;
+    private String validWriteIdList = null;
+
+    public GetPartitionsRequestBuilder(String catName, String dbName, String tblName, boolean withAuth, String user,
+                                       List<String> groupNames, GetProjectionsSpec projectionSpec,
+                                       GetPartitionsFilterSpec filterSpec, List<String> processorCapabilities,
+                                       String processorIdentifier, String validWriteIdList) {
+        this.catName = catName;
+        this.dbName = dbName;
+        this.tblName = tblName;
+        this.withAuth = withAuth;
+        this.user = user;
+        this.groupNames = groupNames;
+        this.projectionSpec = projectionSpec;
+        this.filterSpec = filterSpec;
+        this.processorCapabilities = processorCapabilities;
```

Comment:
the processorCapabilities and the processorIdentifier are meant to be static values for a given instance of HMSClient object. They are not meant to change between different calls. So having it on builder class might be confusing. Should we remove it?

Reply:
These seems harmless to me. I agree they are static, but then they are also public attributes that can be set. Having them in the builder methods is not very different from them being accessed through an object. I have retained them for now. However if you feel strongly about it, I will change it.